compiler/rustc_passes/src/dead.rs

```diff
@@ -726,6 +728,9 @@ impl<'tcx> Visitor<'tcx> for DeadVisitor<'tcx> {
     }
 
     fn visit_item(&mut self, item: &'tcx hir::Item<'tcx>) {
+        if let hir::ItemKind::Mod(..) = item.kind {
```

Comment:
Hm, this looks slightly awkward to me. I wonder if it wouldn't be slightly more straightforward if `visit_mod` was overriden to be empty and then call `walk_mod(visitor, module, module_id)` when entering the visitor in `check_mod_deathness`?

That said, no strong opinion. Feel free to ignore, if you current approach is better.